nncf/common/hook_handle.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import weakref
+from abc import ABC
+from abc import abstractmethod
+from typing import Any, Dict, Union
+
+
+class HookHandleInterface(ABC):
+    """
+    A handle to remove a hook.
+    """
+
+    @property
+    @abstractmethod
+    def hook_id(self) -> str:
+        """
+        Key to use to retrieve handle from the registry.
+        """
+
+    @abstractmethod
+    def remove(self) -> None:
+        """
+        Removes added operation from registered hooks registry if it is possible.
+        """
+
+
+class _HookHandle(HookHandleInterface):
```

Comment:
Why is the private class?

Reply:
I don't share your concerns. I think developers can take responsibility to feed correct parameters in the __init__ method. The main concern that you return `_HookHandle` instance to user in `add_op_to_registry` and it does not look as good story.

Reply:
Done